### PR TITLE
workspace template init commit

### PR DESCRIPTION
**Pull request by Hailong-am**
### Description

workspace template register and consume

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/components/workspace_app.tsx [hunk 1]
```diff
@@ -1,18 +1,39 @@
...
+      workspaceTemplates.push(...templates);
+      for (const template of templates) {
+        const features = templateFeatureMap.get(template.id) || [];
+        features.push(app);
+        templateFeatureMap.set(template.id, features);
+      }
+    })
+  );
+
+  workspaceTemplates = [...new Set(workspaceTemplates)];
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_app.tsx:36]: Seems this `workspaceTemplates` wasn't be used in the workspace, is it only a demo here? Do we need sort by order here? How about separate a hooks to do this?

File: src/plugins/workspace/public/components/workspace_app.tsx [hunk 2]
```diff
@@ -1,18 +1,39 @@
...
 export const WorkspaceApp = ({ appBasePath }: { appBasePath: string }) => {
   const {
-    services: { chrome },
+    services: { chrome, application },
   } = useOpenSearchDashboards();
   const location = useLocation();
+  let workspaceTemplates = [] as WorkspaceTemplate[];
+  const templateFeatureMap = new Map<string, PublicAppInfo[]>();
+  application?.applications$.subscribe((map) =>
+    map.forEach((app) => {
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_app.tsx:25]: How about renaming `map` to `applications`?

File: src/plugins/workspace/public/components/workspace_app.tsx (hunk 1 above)
Comment (Hailong-am) [reply src/plugins/workspace/public/components/workspace_app.tsx:36]: thanks for the comment.Yes, for demo only. Let me extract this logic into a hook.

File: src/plugins/workspace/public/components/workspace_app.tsx (hunk 2 above)
Comment (Hailong-am) [reply src/plugins/workspace/public/components/workspace_app.tsx]: updated

File: src/plugins/workspace/public/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      for (const template of templates) {
+        const features = templateFeatureMap.get(template.id) || [];
+        features.push(app);
+        templateFeatureMap.set(template.id, features);
+      }
+    })
+  );
+
+  workspaceTemplates.sort((a, b) => (a.order || 0) - (b.order || 0));
+  workspaceTemplates = [...new Set(workspaceTemplates)];
```
[the remark this answers is not in the document]
Comment (Hailong-am) [reply src/plugins/workspace/public/hooks.ts]: update as suggested

File: src/plugins/workspace/public/hooks.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ApplicationStart, PublicAppInfo } from 'opensearch-dashboards/public';
+import { WorkspaceTemplate } from '../../../core/types';
+
+export function useWorkspaceTemplate(application: ApplicationStart) {
```
[the remark this answers is not in the document]
Comment (Hailong-am) [reply src/plugins/workspace/public/hooks.ts:11]: thanks for the comment, update as suggested.

**Pull request merged**